src/software/ai/hl/stp/stp.cpp

```diff
@@ -87,7 +94,23 @@ std::vector<std::unique_ptr<Intent>> STP::getIntentsFromCurrentPlay(const World&
 std::vector<std::unique_ptr<Intent>> STP::getIntents(const World& world)
 {
     updateSTPState(world);
-    return getIntentsFromCurrentPlay(world);
+    auto intents = getIntentsFromCurrentPlay(world);
+
+    auto all_tactics = stop_tactics;
+    all_tactics.push_back(goalie_tactic);
+    for (auto tactic : all_tactics)
+    {
+        auto iter = robot_tactic_assignment.find(tactic);
+        if (iter != robot_tactic_assignment.end())
```

Comment:
Can you explain this if statement - why can't `iter` equal `robot_tactic_assignment.end()`?

Reply:
`iter != robot_tactic_assignment.end()` means that the tactic exists in the `robot_tactic_assignment` map